File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;
class Solution {
     public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        for(int a = 0; a < commands.length; a++){
            int i = commands[a][0];
            int j = commands[a][1];
            int k = commands[a][2];
            int[] arr = new int[j - i + 1];
            int n = 0;
            
            for(int b = i - 1; b < j; b++){
                arr[n] = array[b];
                n++;
            }
            Arrays.sort(arr);
            answer[a] = arr[k-1];
        }
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cs
using System;

public class Solution {
    public int[] solution(int[] array, int[,] commands) {
        int n = commands.GetLength(0);
        int[] answer = new int[n];

        for (int idx = 0; idx < n; idx++) {
            int i = commands[idx, 0];
            int j = commands[idx, 1];
            int k = commands[idx, 2];

            int len = j - i + 1;
            int[] subArray = new int[len];
            Array.Copy(array, i - 1, subArray, 0, len);

            // 정렬
            Array.Sort(subArray);

            // k번째 원소 선택
            answer[idx] = subArray[k - 1];
        }

        return answer;
    }
}
